src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -79,26 +82,26 @@ public void drawRoute(final Graphics2D graphics, final RouteDescription routeDes
   /**
    * Draws Points on the Map
    * 
-   * @param graphics The Graphics2D Object being drawn on
-   * @param points The Point array aka the "Joints" to be drawn
+   * @param graphics The {@linkplain Graphics2D} Object being drawn on
+   * @param points The {@linkplain Point} array aka the "Joints" to be drawn
    * @param xOffset The horizontal pixel-difference between the frame and the Map
    * @param yOffset The vertical pixel-difference between the frame and the Map
    * @param jointsize The diameter of the Points being drawn
    * @param scale The scale-factor of the Map
    */
   private void drawJoints(Graphics2D graphics, Point[] points, int xOffset, int yOffset, int jointsize,
       double scale) {
-    for (Point p : points) {
-      graphics.fillOval((int) (((p.x - xOffset) - jointsize / 2) * scale),
-          (int) (((p.y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
+    Point[] newPoints = points.length > 1 ? Arrays.copyOf(points, points.length - 1) : points;
```

Comment:
It's not immediately clear what is happening in this line, and why:
` Point[] newPoints = points.length > 1 ? Arrays.copyOf(points, points.length - 1) : points;`

A comment, or a moving the one line to a function, so the operation would be named,  would be two possibilities to fix